File: datasets/lua-keep-incoder_6b_temp_0.2-1/HumanEval_47_median.lua
-- Return median of elements in the list l.
-- >>> median([3, 1, 2, 4, 5])
-- 3
-- >>> median([-10, 4, 6, 1000, 10, 20])
-- 15.0
local function median(l)
  if #l == 0 then return nil end
  local mid = math.floor(#l / 2)
  if #l % 2 == 0 then
    return l[mid]
  else
    return median(l[mid], l[#l - 1])
  end
end


lu = require('luaunit')

function test_humaneval()
local candidate = median
    lu.assertEquals(candidate({3, 1, 2, 4, 5}), 3)
    lu.assertEquals(candidate({-10, 4, 6, 1000, 10, 20}), 8.0)
    lu.assertEquals(candidate({5}), 5)
    lu.assertEquals(candidate({6, 5}), 5.5)
    lu.assertEquals(candidate({8, 1, 3, 9, 9, 2, 7}), 7)
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_47_median.R
# Return median of elements in the list l.
-- >>> median([3, 1, 2, 4, 5])
-- 3
-- >>> median([-10, 4, 6, 1000, 10, 20])
-- 15.0
median <- function(l)
{
    l <- sort(l)
    n <- length(l)
    if (n %% 2 == 0)
    {
        return((l[n/2] + l[n/2 + 1])/2)
    }
    else
    {
        return(l[floor(n/2) + 1])
    }

test_humaneval <- function() {
candidate <- median
    if(!identical(candidate(c(3, 1, 2, 4, 5)), 3)){quit('no', 1)}
    if(!identical(candidate(c(-10, 4, 6, 1000, 10, 20)), 8.0)){quit('no', 1)}
    if(!identical(candidate(c(5)), 5)){quit('no', 1)}
    if(!identical(candidate(c(6, 5)), 5.5)){quit('no', 1)}
    if(!identical(candidate(c(8, 1, 3, 9, 9, 2, 7)), 7)){quit('no', 1)}
}